lib/pub_sub/railties/active_record.rb

```diff
@@ -5,9 +5,14 @@ def self.included(klass)
         klass.send :include, InstanceMethods
         klass.send :extend, ClassMethods
 
-        klass.after_save :publish_changes, if: -> { changed? }
-        klass.after_destroy :publish_changes
-        klass.after_touch :publish_changes
+        klass.after_save { @__pub_sub_record_unchanged = !changed? }
+        # Not everything executes the `after_save` callback (like touch and destroy),
+        # so use `unless` logic so the uninitialized @__xxx_unchanged is nil, which equals false.
+        klass.after_commit :publish_changes, unless: Proc.new {
+          # return current unchanged status and reset unchanged status to false
+          x, @__pub_sub_record_unchanged = @__pub_sub_record_unchanged, false
```

Comment:
Yeah, I had that originally, butis is sooo many more lines of code to accomplish the same thing...